src/main/java/org/assertj/core/api/ClassBasedNavigableCollectionAssert.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2021 the original author or authors.
+ */
+package org.assertj.core.api;
+
+import static org.assertj.core.util.Preconditions.checkArgument;
+
+import java.lang.reflect.Constructor;
+import java.util.Collection;
+
+/**
+ * Build the Assert instance by reflection.
+ * 
+ * @since 3.21.0
+ */
+//@format:off
+public class ClassBasedNavigableCollectionAssert<SELF extends ClassBasedNavigableCollectionAssert<SELF, ACTUAL, ELEMENT, ELEMENT_ASSERT>, 
```

Comment:
I'm not sure about the concrete use cases for `ClassBasedNavigableCollectionAssert` and `FactoryBasedNavigableCollectionAssert`, and if it makes sense at all to have them. For now, the structure is similar to `ListAssert` and `IterableAssert`.

Reply:
it's to get narrowed type assertions when navigating to elements,ex:
```java
Iterable<String> hobbits = newHashSet("frodo", "sam", "pippin");
assertThat(hobbits, StringAssert.class).first()
                                        .startsWith("fro")
                                        .endsWith("do");
```
This was introduced before instanceof factories which fills a similar purpose.
I don't mind leaving them out if that's possible.

Reply:
Ok, so I'll keep it simple and remove them. We can always add them later if users request them.